torch_em/data/segmentation_dataset.py

```diff
@@ -123,11 +125,22 @@ def _sample_bounding_box(self):
     def _get_sample(self, index):
         if self.raw is None or self.labels is None:
             raise RuntimeError("SegmentationDataset has not been properly deserialized.")
+
         bb = self._sample_bounding_box()
         bb_raw = (slice(None),) + bb if self._with_channels else bb
         bb_labels = (slice(None),) + bb if self._with_label_channels else bb
         raw, labels = self.raw[bb_raw], self.labels[bb_labels]
 
+        # Padding the patch to match the expected input shape.
+        if self.patch_shape is not None and self.with_padding:
+            if any(sh < psh for sh, psh in zip(raw.shape, self.patch_shape)):
```

Comment:
This logic will fail if we have channels. This is also why we use `_ensure_patch_shape` in `ImageCollectionDataset`. I would suggest that we refactor `_ensure_patch_shape` into `torch_em.util` so that we can use this in both datasets.

Reply:
This is updated now!